src/Admin/AddProduct.js

```diff
@@ -1,5 +1,85 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import useStorage from "./models/useStorage";
+import useDatabase from "./models/useDatabase";
+
+import "./admin.css";
+
+// Components
+import ProgressBar from "./components/ProgressBar";
 
 export default function AddProduct() {
-  return <div>Add a product here</div>;
+  const [preview, setPreview] = useState(null);
+  const [selection, setSelection] = useState(null);
+  const [productName, setProductName] = useState("");
+  const types = ["image/png", "image/jpeg"];
+  const [file, setFile] = useState(null);
+  const { progress, url, error } = useStorage(file);
+  const [product, setProduct] = useState(null);
```

Comment:
Most of this seems to be for managing the upload of the file.  It seems like we might be able to do this when submit occurs and skip tracking most of this.

Reply:
I was following a tutorial where the instructor (net ninja) selects a file and uploads to the, pulls the url and then shoots this off to the database. It was for an Instagram type thing so he didn't have any additional details required to the database other than a timestamp and id. This was all done in a single step. However (and mainly because I am trying to do this in my bug english app) I wanted to initially set a preview of the photo in the UI (preview) and only send the photo to FB once we click submit. setFile is the event that triggers the photo to go to storage, but I need to save the file that I want to send (selected) in scope as I can't access the variable it's saved to outside the handleAddImage function.